protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java

```diff
@@ -0,0 +1,254 @@
+               LOG.error("XmpDetector: no config factory, using defaults");   
+            else
+	       System.out.println("XmpDetector: null LOG");
+        }
+    }
+
+    public int getPort()
+    {
+        LOG.debug("XmpDetector: getPort");
+	return xmpPort;
+    }
+
+    public void setPort(int newPort)
+    {
+        LOG.debug("XmpDetector: setPort to "+newPort);
+	xmpPort = newPort;
+    }
+
+    public void setIpMatch(String ipMatch)
+    {
+	LOG.debug("XmpDetector: setIpMatch to "+ipMatch);
+	
+	m_ipMatch = ipMatch;
+    }
+
+    public String getIpMatch()
+    {
+        LOG.debug("XmpDetector: getIpMatch returning '"+m_ipMatch+"'");
+	return m_ipMatch;
+    }
+
+    public int getTimeout()
+    {
+        LOG.debug("XmpDetector: getTimeout returning "+xmpTimeout);
+	return xmpTimeout;
+    }
+
+    public void setTimeout(int newTimeout) 
+    {
+        LOG.debug("XmpDetector: setTimeout to "+newTimeout);
+        xmpTimeout = newTimeout; 
+        sockopts.setConnectTimeout(xmpTimeout);
+    }
+
+    public void dispose()
+    {
+	LOG.debug("XmpDetector: dispose invoked");
+	
+	// dispose of anything like sessions, etc.
+        // no need to dispose SocketOpts
+    }
+
+    @Override
+    public final boolean isServiceDetected(InetAddress address)
+    {
+        XmpSession aSession;
+        XmpMessage aReply;
+        XmpVar[] vars,replyVars;
+
+	LOG.debug("XmpDetector: isServiceDetected starting to query "+address);
+	
+        // try to establish session
+        aSession = new XmpSession(sockopts,address,xmpPort,xmpAuthenUser);
+        if (aSession == null) {
+	   LOG.debug("XmpDetector: null session to "+address);
+	   return false;
+        }
+
+        LOG.debug("XmpDetector: isServiceDetected session established with "+address);
+        // query for core.sysName, core.sysDescr, 
+        // core.sysUpTime, core.xmpdVersion
+        vars = new XmpVar[] {
+	    new XmpVar("core","sysName","","",Xmp.SYNTAX_NULLSYNTAX),
+	    new XmpVar("core","sysDescr","","",Xmp.SYNTAX_NULLSYNTAX),
+	    new XmpVar("core","sysUpTime","","",Xmp.SYNTAX_NULLSYNTAX),
+	    new XmpVar("core","xmpdVersion","","",Xmp.SYNTAX_NULLSYNTAX),
+        };
+      
+        if ((aReply = aSession.queryVars(vars)) == null) {
+            LOG.debug("XmpDetector: isServiceDetected no vars from "+address);
+	    aSession.closeSession();
+            return false;
+        }
+
+        aSession.closeSession();
+
+	// log what we retrieved
+        if ((replyVars = aReply.getMIBVars()) == null) {
+            LOG.debug("XmpDetector: isServiceDetected no replyVars for "+address);
+            return false;
+
+        } /* if replyVars == null */
+
+        LOG.debug("XmpDetector: isServiceDetected "+address+" reports "+
+                   replyVars[0].getValue()+","+
+                   replyVars[1].getValue());
+        LOG.debug("XmpDetector: isServiceDetected true for "+address);
+
+        LOG.debug("XmpDetector: isServiceDetected true for "+address);
```

Comment:
Made the changes and am test compiling the code.  Do you need me to issue another pull request?